lib/rubygems/remote_fetcher.rb

```diff
@@ -234,7 +234,7 @@ def fetch_http(uri, last_modified = nil, head = false, depth = 0)
 
       fetch_http(location, last_modified, head, depth + 1)
     else
-      raise FetchError.new("bad response #{response.message} #{response.code}", uri)
+      raise FetchError.new("bad response #{response.message} #{response.code} #{response.body.strip}", uri)
```

Comment:
This can be suuuuuuuuuuuuper messy IMHO when remote provides like 404 HTML page. Would it make sense to limit this to first X characters at least or scan for friendly response and report back only when recognized format is returned? What's returned by RubyGems.org in here?

Reply:
Referencing a gem that doesn't exist doesn't result in this message being displayed.

I'm not opposed to adding a character limit, though.

```
puppy@orthrus:~/test$ ls -a
./       ../      Gemfile
puppy@orthrus:~/test$ cat Gemfile
source 'https://rubygems.org'

gem 'asdfasdfsadfihgjklsadkjfhakljfhsaf-this-gem-does-not-exist'
puppy@orthrus:~/test$ bundle install
Fetching gem metadata from https://rubygems.org/.
Could not find gem 'asdfasdfsadfihgjklsadkjfhakljfhsaf-this-gem-does-not-exist' in rubygems repository
https://rubygems.org/ or installed locally.
puppy@orthrus:~/test$ 
```

Reply:
There are other gem servers which can return super long 500 HTML page for example. I would suggest to revert this and instead build some kind of "protocol" to let the error message propagate to client. IMHO it is not good idea to propagate licensing error back to clients using HTTP message on gem install. There is no guarantee how RubyGems handles HTTP errors and IMHO it should not be mis-used to handing over messages to client.

Reply:
I would be happy to keep this output, but only in kind of "verbose" mode and that is most likely against the original need of @adrianthedev.

Reply:
We have also sometimes needed this in Bundler, for example when rate-limiting certain requests. I like the idea of this being officially supported, but I agree that an unlimited message size is messy and unhelpful.

Reply:
I was thinking how to get this done with raw HTTP without inventing custom thingies like special payload or headers. What about to check for custom content-type (per https://en.wikipedia.org/wiki/Media_type#Vendor_tree like `application/vnd.rubygems.text`) and print the content on non 4xx and 5xx responses?

Reply:
That sounds good for "success", but I would also like a way to print errors for rate limiting (429) and server errors (5xx), so we can provide additional information or explanations if desired.

Reply:
> That sounds good for "success", but I would also like a way to print errors for rate limiting (429) and server errors (5xx), so we can provide additional information or explanations if desired.

I'm not sure I do follow. That is the idea, if response has special recognized content-type, print out the response content, but only for 4xx and 5xx responses.

Reply:
Oh, sorry, I misunderstood! I thought you were saying only responses _not_ 4xx or 5xx. 🙃 Sounds good to me.

Reply:
@adrianthedev would that work for you? ^ Are you in control of returned HTTP headers for 403 on your gemserver?

Reply:
Yes @simi, we are in control of the returned HTTP headers. Our gem server is operated by a sinatra app.

Not completely sure how to send the message. 
Is it something like this?

```ruby
halt 403, {'Content-Type' => 'application/vnd.rubygems.text'}, { message: 'some text' }.to_json
```

Our message is something like this. That's 164 characters. We might play around with it so the number of chars might increase.

```
It seems that you don't have a valid GEM_SERVER_TOKEN for Avo (https://bundler.dev/). Please read this for more information
https://docs.avohq.io/3.0/some-path-here
```

Reply:
Did I understand correctly that we'd rather revert this PR and go with the cleaner solution that @simi proposed?

Reply:
Yes, if possible, don't release this. 🙏